programs/stake/src/stake_state.rs

```diff
@@ -431,6 +431,13 @@ impl Delegation {
             None
         }
     }
+
+    fn can_merge_with(&self, source: &Self) -> bool {
+        self.voter_pubkey == source.voter_pubkey
+            && (self.warmup_cooldown_rate - source.warmup_cooldown_rate).abs() < f64::EPSILON
```

Comment:
you also want to verify that they were activated at the same time or are fully activated?

Reply:
Hey! Thanks for taking a peek! :slightly_smiling_face: 

I wasn't planning on handling the transient states.  The fully-activated check is down here https://github.com/solana-labs/solana/pull/13712/files#diff-ab7997cf6770a6fae7b258e433b5dd8c33afa30431eab212552bc7f2b7b38db1R1164